eng/packaging.targets

```diff
@@ -54,29 +76,55 @@
              Targets="Build" />
   </Target>
 
-  <Target Name="AddRuntimeSpecificAssemblies"
-          DependsOnTargets="BuiltProjectOutputGroup;SatelliteDllsProjectOutputGroup;$(TargetsForTfmSpecificBuildOutput)"
-          Condition="'$(TargetsAnyOS)' != 'true' and '$(ExcludeFromPackage)' == ''">
+  <Target Name="AddDocumentationFileToPackage">
+    <PropertyGroup>
+      <!-- Search for the documentation file in the intellisense package and otherwise pick up the generated one. -->
+      <_documentationFileToPackage Condition="Exists('$(XmlDocFileRoot)1033\$(TargetName).xml')">$(XmlDocFileRoot)1033\$(TargetName).xml</_documentationFileToPackage>
+      <_documentationFileToPackage Condition="'$(_documentationFileToPackage)' == '' and Exists('$(DocumentationFile)')">$(DocumentationFile)</_documentationFileToPackage>
+    </PropertyGroup>
+    <ItemGroup Condition="'$(_documentationFileToPackage)' != ''">
+      <!-- Place the documentation file into the lib or runtimes/lib folder depending on if the build is runtime agnostic or specific. -->
+      <TfmSpecificPackageFile Include="$(_documentationFileToPackage)"
+                              PackagePath="$([MSBuild]::ValueOrDefault('$(BuildOutputTargetFolder)', 'lib'))/$(TargetFrameworkWithoutSuffix)"
+                              Condition="'$(TargetsAnyOS)' == 'true'" />
+      <TfmRuntimeSpecificPackageFile Include="$(_documentationFileToPackage)"
+                                     Condition="'$(TargetsAnyOS)' != 'true'" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddRuntimeSpecificFilesToPackage"
+          DependsOnTargets="BuiltProjectOutputGroup;
+                            SatelliteDllsProjectOutputGroup;
+                            $(TargetsForTfmSpecificBuildOutput);
+                            AddDocumentationFileToPackage"
+          Condition="'$(TargetsAnyOS)' != 'true'">
     <ItemGroup>
-      <TfmSpecificPackageFile Include="@(SatelliteDllsProjectOutputGroupOutput);
-                            @(BuiltProjectOutputGroupOutput)">
-        <PackagePath>runtimes/$(_runtimeOS)/lib/$(TargetFrameworkWithoutSuffix)</PackagePath>
-      </TfmSpecificPackageFile>
+      <TfmRuntimeSpecificPackageFile Include="@(BuiltProjectOutputGroupOutput);
+                                              @(SatelliteDllsProjectOutputGroupOutput)" />
+      <TfmSpecificPackageFile Include="@(TfmRuntimeSpecificPackageFile)"
+                              PackagePath="runtimes/$(PackageTargetRuntime)/lib/$(TargetFrameworkWithoutSuffix)" />
+      <!-- Copy runtime specific assemblies into the rid agnostic 'lib' folder if the ridless tfm doesn't exist.
+           The assumption holds that a ridless tfm doesn't follow a '-' charater. -->
+      <TfmSpecificPackageFile Include="@(TfmRuntimeSpecificPackageFile)"
```

Comment:
why are we doing this ?

Reply:
which runtime assembly it is going to choose? I think if a package doesn't provide a rid agnostic tfm it should as that is a wrong authored package and if it can't run rid agnostic it should throw?

Reply:
That's what the pkgproj infrastructure already does today. I refined the comment by adding a note about the NuGet error this avoids: NU5128

Reply:
I agree with what others have said, we shouldn't be doing this.  Projects need to have a RID-less TFM to be packed.

Reply:
@ericstj the existing pkgproj infrastructure is doing the exact same thing. As an example, System.IO.Ports doesn't define a ridless tfm today but the package assets get copied into both the runtimes and the lib folder: https://github.com/dotnet/runtime/blob/b056b7b6a1eca5171e42f697a0b3d3c60d1fc048/src/libraries/System.IO.Ports/src/System.IO.Ports.csproj#L7.

The goal is to maintain the existing experience and not introduce new build targets. Also I don't think that @safern meant that he is against the change but just answered @Anipik's question and provided more context.

Reply:
This support is only there for 10 .NET Framework libs which have a rid configuration but not a ridless one. The rid specific configuration need to be kept for these as otherwise the `netstandard2.0-windows` configuration would win over the .NET Framework one.

I don't see any reason why a library that targets .NET Framework needs both a rid specific and a ridless tfm.

Reply:
The pkgproj infra only did this for .NETFramework, and was to ensure that folks who targeted .NETFramework with a RID didn't get a .NETStandard RID-specific asset.  I believe classic CSProj defines a RID [by default](https://github.com/dotnet/NuGet.BuildTasks/blob/b2f0c162994237c4a9c23af79d95f79847499ebd/src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.targets#L113) for .NETFramework projects so it's more common than you'd think.
Here's how pkgproj did this: https://github.com/dotnet/arcade/blob/35a36a8ec705945b20275a7d63478affb396ff89/src/Microsoft.DotNet.Build.Tasks.Packaging/src/GetPackageDestination.cs#L149-L162
I'd be OK if we constrained this workaround to only RID-specific configurations of .NETFramework.

Reply:
> What do you propose so that classic .NETFramework projects don't get the RID-specific netstandard asset?

I was only talking about build configurations. It's clear that if a runtimes/win/lib/netstandard2.0 asset is present that a runtimes/win/lib/net461 one needs to exist as well if the .NET Standard asset shouldn't be consumed.

I propose to keep the logic as is but condition it on '$(TargetFrameworkIdentifier)' == '.NETFramework' so that it stays .NET Framework specific, exactly like in the pkgproj system. It sounded like you would be ok with that.